packages/amplication-server/src/core/subscription/subscription.service.ts

```diff
@@ -199,26 +211,41 @@ export class SubscriptionService {
     });
   }
 
+  async trackUpgradeCompletedEvent(
+    plan: EnumSubscriptionPlan,
+    userId: string,
+    workspaceId: string
+  ): Promise<void> {
+    if (plan !== EnumSubscriptionPlan.Free && userId) {
+      await this.analyticsService.track({
+        userId: userId,
+        properties: {
+          workspaceId: workspaceId,
+        },
+        event: EnumEventType.WorkspacePlanUpgradeCompleted,
+      });
+    }
+  }
+
   async handleUpdateSubscriptionStatusEvent(
     updateStatusDto: UpdateStatusDto
   ): Promise<void> {
     switch (updateStatusDto.type) {
       case "subscription.created": {
         const createSubscriptionInput =
           this.mapUpdateStatusDtoToCreateSubscriptionInput(updateStatusDto);
-        await this.create(updateStatusDto.id, createSubscriptionInput);
-        const userId = updateStatusDto.metadata?.userId;
-        if (
-          createSubscriptionInput.plan !== EnumSubscriptionPlan.Free &&
-          userId
-        ) {
-          await this.analyticsService.track({
-            userId: userId,
-            properties: {
-              workspaceId: updateStatusDto.customer.id,
-            },
-            event: EnumEventType.WorkspacePlanUpgradeCompleted,
-          });
+        const subs = await this.getSubscriptionById(updateStatusDto.id);
+        if (subs) {
+          await this.create(updateStatusDto.id, createSubscriptionInput);
+          await this.trackUpgradeCompletedEvent(
+            createSubscriptionInput.plan,
+            updateStatusDto.metadata?.userId,
+            updateStatusDto.customer.id
+          );
+        } else {
+          this.logger.warn(
+            `Trying to create an existing subscription with id: ${updateStatusDto.id}`
```

Comment:
Let's add the updateDto to the log that it might give us some more insight about the event.

Reply:
Done